### PR TITLE
[Bug fix] Prevent exposing API key secret in stdout logging

### PR DESCRIPTION
**Pull request by paulegradie**
# Background

A customer revealed to us that we are exposing an api key secret during a call to `Write-Verbose` in one of our configuration scripts for `cTentancleAgent`. This was due to a redundant bit of logging that was not passed through our secret scrubber function `Get-MaskedOutput`. Removing the logging line resolves this problem while leaving the masked logging in tact.

# Results

Duplicate logging no longer occurs. 

# How to review this PR
 - Quality ✔️ 

We current provide test coverage on the methods used to mask outputs, however we do not text stdout via assent.

This issue has been reported to the School Security for their ongoing secrets related investigations.

### CODE REVIEW

File: .github/pull_request_template.md
Comment (matt-richardson) [inline .github/pull_request_template.md:1]: Can this be done as a separate PR... I think there's discussion we want to have around it first.

**Review by matt-richardson — CHANGES_REQUESTED** [review]
Nice one. Should do what it says on the tin.
I'd like to have a discussion around the pull request template before we merge that in.
Also, I'd like to see us fix up the issues around the "not catching all the permutations" that we spotted on the way.

File: .github/pull_request_template.md
Comment (paulegradie) [reply .github/pull_request_template.md:1]: Sure, we can leave that to another PR.

Comment (paulegradie) [reply .github/pull_request_template.md:1]: Ah woops - just a heads up on this - I intended to provide a barebones template. Its updated now to that, but I will strip this out into a separate PR momentarily. 👍

File: OctopusDSC/DSCResources/cTentacleAgent/cTentacleAgent.psm1
Comment (paulegradie) [inline OctopusDSC/DSCResources/cTentacleAgent/cTentacleAgent.psm1:1281]: This is the main fix - this line is redundant and doesn't pass the arguments through the scrubber.

File: OctopusDSC/OctopusDSCHelpers.ps1
Comment (paulegradie) [inline OctopusDSC/OctopusDSCHelpers.ps1:161]: We felt it was prudent to improve the logic around the masking function. Note the edge case clause (the else block).

File: OctopusDSC/OctopusDSCHelpers.ps1
Comment (paulegradie) [inline OctopusDSC/OctopusDSCHelpers.ps1:187]: Rewriting the function masker helped make it clear that much of this code was redundant, so it has been removed. Args provided to a call to Write-Verbose (or other logging) should always be passed through the masking function.

File: README.md
Comment (paulegradie) [inline README.md:27]: Following the guide, I noticed some of these links no longer work. (the vagrant.io tried to open from within the github domain).

We could consider a little more detail concerning TeamCity env variables.

**Review by matt-richardson — APPROVED** [review]
LGTM. Nice one - great to see some `TODO` comments disappear.

**Comment by paulegradie** [comment]
[API key printed via DSC](https://trello.com/c/40cA3I65/4098-api-key-printed-via-dsc)

**Comment by matt-richardson** [comment]
You'd have to use the audit log to review changes made via that api key.
Best bet is to also change the api key, just in case.

**Comment by attritionorg** [comment]
@paulegradie Sorry for the late question, but what are the default permissions for where this was logged? Is this a proactive "just in case" fix and the file was e.g. 600 or equiv, or was the file world-readable? Thanks!

**Pull request merged**
